File: scripts/audit_colors.js
const fs = require('fs');
const path = require('path');

const PURPLE_PATTERNS = [
  /#a855f7/gi,
  /#8a2be2/gi,
  /#9370db/gi,
  /#c084fc/gi,
  /#8b5cf6/gi,
  /#7c3aed/gi,
  /#6d28d9/gi,
  /#bf5af2/gi,
  /#d500f9/gi,
  /#e879f9/gi,
  /#d946ef/gi,
  /#a21caf/gi,
  /#a168f9/gi,
  /#a15cff/gi,
  /rgba\(\s*168\s*,\s*85\s*,\s*247/gi,
  /rgba\(\s*138\s*,\s*43\s*,\s*226/gi,
  /rgba\(\s*139\s*,\s*92\s*,\s*246/gi
];

const TARGET_DIRS = ['css', 'JS'];
const TARGET_FILES = ['index.html'];

let totalMatches = 0;

function checkFile(filePath) {
  const content = fs.readFileSync(filePath, 'utf8');
  const lines = content.split('\n');
  const fileMatches = [];

  lines.forEach((line, index) => {
    for (const pattern of PURPLE_PATTERNS) {
      if (pattern.test(line)) {
        fileMatches.push({ lineNum: index + 1, text: line.trim() });
        break;
      }
    }
  });

  if (fileMatches.length > 0) {
    console.log(`\n📄 ${filePath} (${fileMatches.length} matches):`);
    fileMatches.forEach(m => console.log(`  L${m.lineNum}: ${m.text}`));
    totalMatches += fileMatches.length;
  }
}

function walkDir(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      if (file !== 'node_modules' && file !== '.git') walkDir(fullPath);
    } else if (file.endsWith('.css') || file.endsWith('.js')) {
      checkFile(fullPath);
    }
  }
}

console.log('--- SCANNING CODEBASE FOR PURPLE/NEON ARTIFACTS ---');
TARGET_DIRS.forEach(d => {
  if (fs.existsSync(d)) walkDir(d);
});
TARGET_FILES.forEach(f => {
  if (fs.existsSync(f)) checkFile(f);
});

console.log(`\nTOTAL PURPLE MATCHES FOUND: ${totalMatches}`);
process.exit(totalMatches > 0 ? 1 : 0);

File: scripts/audit_colors.py
import os
import re
import sys

PURPLE_PATTERNS = [
    re.compile(r'#a855f7', re.I),
    re.compile(r'#8a2be2', re.I),
    re.compile(r'#9370db', re.I),
    re.compile(r'#c084fc', re.I),
    re.compile(r'#8b5cf6', re.I),
    re.compile(r'#7c3aed', re.I),
    re.compile(r'#6d28d9', re.I),
    re.compile(r'#bf5af2', re.I),
    re.compile(r'#d500f9', re.I),
    re.compile(r'#e879f9', re.I),
    re.compile(r'#d946ef', re.I),
    re.compile(r'#a21caf', re.I),
    re.compile(r'#a168f9', re.I),
    re.compile(r'#a15cff', re.I),
    re.compile(r'rgba\(\s*168\s*,\s*85\s*,\s*247', re.I),
    re.compile(r'rgba\(\s*138\s*,\s*43\s*,\s*226', re.I),
    re.compile(r'rgba\(\s*139\s*,\s*92\s*,\s*246', re.I),
]

TARGET_DIRS = ['css', 'JS']
TARGET_FILES = ['index.html']

total_matches = 0

def check_file(file_path):
    global total_matches
    try:
        with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
            lines = f.readlines()
    except Exception as e:
        return

    matches = []
    for idx, line in enumerate(lines):
        for pat in PURPLE_PATTERNS:
            if pat.search(line):
                matches.append((idx + 1, line.strip()))
                break

    if matches:
        print(f"\n[FILE] {file_path} ({len(matches)} matches):")
        for line_num, text in matches:
            safe_text = text.encode('ascii', 'replace').decode('ascii')
            print(f"  L{line_num}: {safe_text[:120]}")
        total_matches += len(matches)

for d in TARGET_DIRS:
    if os.path.exists(d):
        for root, _, files in os.walk(d):
            for file in files:
                if file.endswith('.css') or file.endswith('.js'):
                    check_file(os.path.join(root, file))

for f in TARGET_FILES:
    if os.path.exists(f):
        check_file(f)

print(f"\nTOTAL PURPLE MATCHES FOUND: {total_matches}")
if total_matches > 0:
    sys.exit(1)
else:
    sys.exit(0)
